haystack/preview/components/embedders/gradient_text_embedder.py

```diff
@@ -0,0 +1,80 @@
+from typing import Any, Dict, List, Optional
+
+from haystack.preview import component, default_to_dict
+from haystack.preview.lazy_imports import LazyImport
+
+with LazyImport(message="Run 'pip install gradientai'") as gradientai_import:
+    from gradientai import Gradient
+
+
+@component
+class GradientTextEmbedder:
+    """
+    A component for embedding strings using models hosted on Gradient AI (https://gradient.ai).
+    """
+
+    def __init__(
+        self,
+        *,
+        model_name: str = "bge-large",
+        access_token: Optional[str] = None,
+        workspace_id: Optional[str] = None,
+        host: Optional[str] = None,
+    ) -> None:
+        """
+        Create a GradientTextEmbedder component.
+
+        ```python
+        p = Pipeline()
+        embedder = GradientTextEmbedder(access_token=gradient_access_token)
+        p.add_component(instance=embedder, name="text_embedder")
+        p.add_component(instance=InMemoryEmbeddingRetriever(document_store=InMemoryDocumentStore()), name="retriever")
+        p.connect("text_embedder", "retriever")
+        p.run("embed me!!!")
+        ```
+
+        :param model_name: The name of the model to use.
+        :param access_token: The Gradient access token. If not provided it's read from the environment
+                             variable GRADIENT_ACCESS_TOKEN.
+        :param workspace_id: The Gradient workspace ID. If not provided it's read from the environment
+                             variable GRADIENT_WORKSPACE_ID.
+        :param host: The Gradient host. By default it uses https://api.gradient.ai/.
+        """
+        self._host = host
```

Comment:
The first line here should be `gradientai_import.check()`, to make sure the dependencies are in place when you initialize this component.

Reply:
done!